WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/WebView.swift

```diff
@@ -26,6 +28,12 @@ struct WebView: UIViewRepresentable {
 
     private let credentials = ServiceLocator.stores.sessionManager.defaultCredentials
 
+    init(isPresented: Binding<Bool>, url: URL, previewDevice: Binding<ThemesPreviewView.PreviewDevice?> = .constant(nil)) {
```

Comment:
The type `ThemesPreviewView` is specific to themes preview only, so putting it in a shared view is not a good idea. Consider adding a callback closure for the `didCommit` delegate method and handle the preview at the call site to keep this `WebView` generic.

```suggestion
    init(isPresented: Binding<Bool>, url: URL, onCommit: (() -> Void)? = nil) {
```

Reply:
Thanks for the suggestion, as you also know yesterday this is the area I wasn't sure about how to structure the code.

I added some change on https://github.com/woocommerce/woocommerce-ios/pull/11452/commits/67ae160dbe49b43ad13e3b720539e28df7c33cd4